api/services/RichText.js

```diff
@@ -1,54 +1,155 @@
-        url = Frontend.Route.tokenLogin(recipient, token, url)
-      }
+export function processHTML (contentHTML, providedInsaneOptions) {
+  // NOTE: Will probably fail silently if bad content sent
+  if (!contentHTML) return ''
+
+  const autolinkedHTML = Autolinker.link(contentHTML, { className: 'linkified' })
+  const dom = getDOM(autolinkedHTML)
+
+  forEach(el => {
+    // Remove all `target` attributes for anchors  Concatenate long link text appending "…"
+    el.removeAttribute('target')
+
+    // Add `hylo-link` to internal links
+    if ((el.getAttribute('href') && el.getAttribute('href').match(HYLO_URL_REGEX))) {
+      el.className = 'hylo-link'
     }
 
-    $el.attr('href', url)
-  })
+    // This currently has to be reversed by the TipTap by referencing the href on edit
+    if (el.textContent.length > MAX_LINK_LENGTH) {
+      el.innerHTML = `${el.textContent.slice(0, MAX_LINK_LENGTH)}…`
+    }
+
+    // Normalize legacy Mentions
+    if (
+      el.getAttribute('data-entity-type') === 'mention'||
+      el.getAttribute('data-user-id')
+    ) {
+      const newSpanElement = dom.createElement('span')
+
+      newSpanElement.className = 'mention'
+      newSpanElement.setAttribute('data-type', 'mention')
+      newSpanElement.setAttribute('data-id', el.getAttribute('data-user-id'))
+      newSpanElement.setAttribute('data-label', el.textContent)
+      newSpanElement.innerHTML = el.innerHTML
+
+      el.parentNode.replaceChild(newSpanElement, el)
+
+      return
+    }
+
+    // Normalize legacy Topics
+    if (
+      el.getAttribute('data-entity-type') === '#mention' ||
+      (!el.getAttribute('href') && el.textContent[0] === '#')
+    ) {
+      const newSpanElement = dom.createElement('span')
+
+      newSpanElement.className = 'topic'
+      newSpanElement.setAttribute('data-type', 'topic')
+      newSpanElement.setAttribute('data-id', el.textContent?.slice(1))
+      newSpanElement.setAttribute('data-label', el.textContent)
+      newSpanElement.innerHTML = el.innerHTML
+
+      el.parentNode.replaceChild(newSpanElement, el)
 
-  return $.html()
+      return
+    }
+  }, dom.querySelectorAll('a'))
+  
+  // Always sanitize on output, but only once and only here
+  const  santizedHTML = insane(
+    dom.querySelector('body').innerHTML,
+    TextHelpers.insaneOptions(providedInsaneOptions)
+  )
+
+  return santizedHTML
 }
 
 /*
-Returns a set of unique IDs for any mention members
-found in the provided HTML
 
-Used for generating notifications
+Prepares content for HTML Email delivery
+
+- Always make sure `processHTML` was ran first, this is done
+  in `Post#details()` and `Comment#text()`
+
+- Links will be generated with `/all` if a `groupSlug` is not passed
+
+- This same logic is handled dynamically on Web in `ClickCatcher`,
+  and on Mobile in the `HyloHTML` component.
+
 */
-export const getUserMentions = html => {
-  if (!html) return []
+export function qualifyLinks (processedHTML, groupSlug) {
```

Comment:
This is used for generating links from  in the HTML content for emails. This is the only place in the backend which will generate URLs scoped to a group. Currently everywhere this is ran on the backend a `groupSlug` is provided and pretty much always it's first group on the Post... as we do on both front-ends.